src/geo/map/cartodb-layer.js

```diff
@@ -120,6 +120,26 @@ var CartoDBLayer = LayerModelBase.extend({
     var meta = this.get('meta');
     var stats = meta && meta.stats;
     return stats && stats.estimatedFeatureCount;
+  },
+
+  getSourceId: function () {
+    var source = this.getSource();
+    return source && source.id;
+  },
+
+  getSource: function () {
+    return this.get('source');
+  },
+
+  setSource: function (source, options) {
+    this.set('source', source, options);
+  },
+
+  update: function (attrs) {
+    if (attrs.source) {
```

Comment:
Curious about one thing here. If `source` is set to `null` or `undefined` the error won't trigger but Backbone would set it to that value. That's the same as setting a new value.

I could be wrong but it's worth to check it and use `hasOwnProperty` better than Backbone's `has` (that answers no if it's null or undefined).